azavea/iow-boundary-tool
# Add utility model and development users

## Overview

- Define the `Utility` model
- Require contributors to be associated with a utility
- Enable editing in the Django admin
- Add a `resetdb` script that populates a test utility and test users
- Default new utilities to Raleigh

Closes #64 
Closes #65

Test users:
a1@azavea.com/password
v1@azavea.com/password
c1@azavea.com/password

### Demo

After running `scripts/setup`, which runs `scripts/resetdb`, you have three development users and a test utility:

<img width="819" alt="three-users-and-utility" src="https://user-images.githubusercontent.com/38668450/191043884-0ea0ee89-286e-45fa-b5b4-e10f6b68efa3.png">

<img width="1058" alt="raleigh-utility" src="https://user-images.githubusercontent.com/38668450/191047220-de1ca539-28d6-4007-9774-f74a1dee5c4b.png">




The utility is required when editing contributors:

<img width="846" alt="requirement" src="https://user-images.githubusercontent.com/38668450/191043943-70253036-b0c1-47ae-9116-2786e249aae1.png">


### Notes

- This PR is currently based on #72. When that is merged, I'll rebase on `develop`.
- My understanding of `pwsid` is that it's alphanumeric, see: https://ordspub.epa.gov/ords/sfdw/f?p=108:34:0::NO:34:P34_FISCAL_YEAR,P34_QUARTER:2016,4

## Testing Instructions

- `scripts/update --docker` (or `resetdb`)
- `scripts/server`
- View User edit page in Django admin: http://localhost:8181/admin/api/user and log in as a1@azavea.com/password
  - verify user and utility exist
  - verify user to utility is many to many
  - verify you cannot edit or create contributors without an associated utility
  - verify you can create other users with or without a utility
  - verify only a1@azavea.com is a superuser
- View Utility edit page in Django admin: http://localhost:8181/admin/api/utility
  - verify Azavea Test Utility exists, centered on Raleigh
- Verify `scripts/resetdb` resets your dev database to the default

## Checklist
